TwipsterApp/UserEditingWindow.xaml.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Linq;
+using System.Windows;
+using TwipsterApp.Data;
+using TwipsterApp.Models;
+
+namespace TwipsterApp
+{
+    /// <summary>
+    /// Interaction logic for UserEditingWindow.xaml
+    /// </summary>
+    public partial class UserEditingWindow : Window
+    {
+        public UserEditingWindow()
+        {
+            InitializeComponent();
+        }
+
+        private void OnWindowLoaded(object sender, RoutedEventArgs e)
+        {
+            LoginTexBox.Text = CurrentUserModel.CurrentUser.Login;
+            NameTexBox.Text = CurrentUserModel.CurrentUser.Name;
+            SurnameTexBox.Text = CurrentUserModel.CurrentUser.Surname;
+            DateOfBirthPicker.DisplayDate = CurrentUserModel.CurrentUser.BirthDate;
+        }
+
+
+        private void SaveChangesButton_Click(object sender, RoutedEventArgs e)
+        {
+            using var context = new TwipsterDbContext();
+            try
+            {
+                var currentUserEntity = context.Users.Single(x => x.Login == CurrentUserModel.CurrentUser.Login);
+
+                currentUserEntity.Login = LoginTexBox.Text;
+                currentUserEntity.Name = NameTexBox.Text;
+                currentUserEntity.Surname = SurnameTexBox.Text;
+                currentUserEntity.BirthDate = DateOfBirthPicker.SelectedDate.Value;
+
+                context.SaveChanges();
+                PutCurrentUserInformationToTextBoxt(currentUserEntity);
+
+                Close();
+            } catch (Exception x)
+            {
+                MessageBox.Show(x.Message + "\n Check provided information.");
+            }
+        }
+
+        private void OnCancelButtonClicked(object sender, RoutedEventArgs e)
+        {
+            Close();
+        }
+
+        private void PutCurrentUserInformationToTextBoxt(User currentUserEntity)
+        {
+            var twipsterMainWindowOpened = Application.Current.Windows
```

Comment:
Pass TwipsterMainWindow to ctor argument. And use it from argument. Don't search by RAM.